src/mprispropertiesadaptor.cpp

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (C) 2015-2023 Jolla Ltd.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#include "mprisplayer_p.h"
+#include "mprispropertiesadaptor_p.h"
+
+using namespace Amber;
+
+/*
+ * Implementation of adaptor class MprisPlayerAdaptor
+ */
+
+MprisPropertiesAdaptor::MprisPropertiesAdaptor(MprisPlayerPrivate *parent)
+    : QDBusAbstractAdaptor(parent)
+    , m_playerPrivate(parent)
+{
+    setAutoRelaySignals(true);
+}
+
+MprisPropertiesAdaptor::~MprisPropertiesAdaptor()
+{
+}
+
+void MprisPropertiesAdaptor::ReplyPropertyNotFoundError(const QString &interface_name, const QString &property_name)
+{
+    // The property was not found
+    m_playerPrivate->sendErrorReply(QDBusError::UnknownProperty,
+                                    QString::fromLatin1("Property %1%2%3 was not found in object %4")
```

Comment:
I guess QStringLiteral would be the best choice here.

Reply:
Yes, I guess you're right: the choice here is between an extra run-time Latin1 -> UTF-16 conversion, and doubling the string size in the binary.

I think this approach here is okay; it's used quite widely elsewhere, e.g. in Qt itself